File: fix_dashboards.py
import re

def fix_ceo_cfo_dashboards(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()

    # Fix videographer role check
    content = content.replace("e.role.toLowerCase() == 'videographer'", "e.role.toLowerCase().contains('videographer')")

    # In ceo_dashboard, the updateClient call was broken by me, let's restore it
    if 'ceo_dashboard.dart' in filename:
        broken_button = """                ElevatedButton(
                  style: ElevatedButton.styleFrom(backgroundColor: SageColors.primary),
              );
            },
          );"""
        
        fixed_button = """                ElevatedButton(
                  style: ElevatedButton.styleFrom(backgroundColor: SageColors.primary),
                  onPressed: () {
                    context.read<AppState>().updateClient(
                      c.id,
                      name: nameCtrl.text,
                      contactName: contactNameCtrl.text,
                      contactEmail: contactEmailCtrl.text,
                      contactPhone: contactPhoneCtrl.text,
                      contactAddress: contactAddressCtrl.text,
                      contactWebsite: contactWebsiteCtrl.text,
                      monthlyPayable: double.tryParse(payableCtrl.text),
                      packageType: packageType,
                      contractPeriod: contractPeriod,
                      conversionProbability: conversionProbability,
                      retentionHealth: retentionHealth,
                      nextDueDate: dueDateCtrl.text,
                      paymentsDue: int.tryParse(pendingMonthsCtrl.text),
                      weeklyReels: int.tryParse(reelsCtrl.text),
                      weeklyPosts: int.tryParse(postsCtrl.text),
                      weeklyCarousels: int.tryParse(carouselsCtrl.text),
                      weeklyStories: int.tryParse(storiesCtrl.text),
                      campaigns: int.tryParse(campaignsCtrl.text),
                      campaignReach: campaignReachCtrl.text,
                      assignedVideographerId: assignedVideographerId,
                      sessionRate: double.tryParse(sessionRateCtrl.text),
                      postRequirements: guidelinesCtrl.text,
                    );
                    Navigator.pop(ctx);
                  },
                  child: const Text("SAVE"),
                ),
              ],
            );
          },
        );"""
        content = content.replace(broken_button, fixed_button)

    with open(filename, 'w', encoding='utf-8') as f:
        f.write(content)

def fix_video_dashboard(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    content = re.sub(r"final dStr = '--';\s*final sessions = mySessionTasks\.where\(\(t\) \{\s*final dd = t\.deadline;\s*return '--' == dStr;\s*\}\)\.toList\(\);",
                     r"final sessions = mySessionTasks.where((t) {\n                final dd = t.deadline;\n                return dd.year == date.year && dd.month == date.month && dd.day == date.day;\n              }).toList();", content)
    
    content = re.sub(r"final dStr = '--';\s*final sessions = state\.tasks\.where\(\(t\) \{\s*if \(t\.assignedTo != persona\.id \|\| t\.taskType != 'Session'\) return false;\s*final dd = t\.deadline;\s*return '--' == dStr;\s*\}\)\.toList\(\);",
                     r"final sessions = state.tasks.where((t) {\n      if (t.assignedTo != persona.id || t.taskType != 'Session') return false;\n      final dd = t.deadline;\n      return dd.year == d.year && dd.month == d.month && dd.day == d.day;\n    }).toList();", content)

    content = content.replace("final dateLabel = '  ';", "final dateLabel = '${d.day} ${months[d.month-1]} ${d.year}';")

    with open(filename, 'w', encoding='utf-8') as f:
        f.write(content)

fix_ceo_cfo_dashboards('lib/screens/ceo_dashboard.dart')
fix_ceo_cfo_dashboards('lib/screens/cofounder_dashboard.dart')
fix_ceo_cfo_dashboards('lib/screens/cofounder_dashboard_recovered.dart')
fix_video_dashboard('lib/screens/videographer_dashboard.dart')
print('Done!')

File: fix_dashboards.dart
import 'dart:io';

String extractMethod(String content, String methodName) {
  int startIndex = content.indexOf(methodName);
  if (startIndex == -1) return '';
  int braceCount = 0;
  bool started = false;
  int endIndex = -1;
  for (int i = startIndex; i < content.length; i++) {
    if (content[i] == '{') {
      started = true;
      braceCount++;
    } else if (content[i] == '}') {
      braceCount--;
    }
    if (started && braceCount == 0) {
      endIndex = i + 1;
      break;
    }
  }
  if (endIndex != -1) {
    return content.substring(startIndex, endIndex);
  }
  return '';
}

void main() {
  // 1. Fix CEO dashboard colors
  final ceoFile = File('lib/screens/ceo_dashboard.dart');
  String ceoContent = ceoFile.readAsStringSync();
  ceoContent = ceoContent.replaceAll('SageColors.green', 'Colors.green');
  ceoContent = ceoContent.replaceAll('SageColors.red', 'Colors.red');
  ceoFile.writeAsStringSync(ceoContent);
  print('Fixed CEO colors');

  // 2. Re-copy CFO dashboard
  final cfoFile = File('lib/screens/cofounder_dashboard.dart');
  final tempCfoFile = File('temp_cfo_client.txt');
  String cfoContent = tempCfoFile.readAsStringSync();

  // 3. Extract and append tabs from dialog_extract.txt
  final extractFile = File('dialog_extract.txt');
  String extractContent = extractFile.readAsStringSync();

  // Find where _buildPersonnelTab starts in dialog_extract
  int personnelIdx = extractContent.indexOf('Widget _buildPersonnelTab()');
  if (personnelIdx != -1) {
    // Append the _showTeamForm and _taskSubTab variables which were missed
    String variables = '''
  bool _showTeamForm = false;
  String _taskSubTab = 'ALL';
''';
    // Remove the trailing '}' of the class before appending
    int lastBrace = cfoContent.lastIndexOf('}');
    if (lastBrace != -1) {
      cfoContent = cfoContent.substring(0, lastBrace);
    }

    // Append variables and the rest of dialog_extract
    cfoContent += '\n' + variables + extractContent.substring(personnelIdx);
  }

  // Ensure it doesn't have an extra '}' or is missing one.
  // dialog_extract.txt was truncated before the last '}'.
  cfoContent += '\n}\n';

  // Write the file so far
  cfoFile.writeAsStringSync(cfoContent);

  // 4. Extract client dialogs from dialog_full.txt
  final fullFile = File('dialog_full.txt');
  String editClientStr = extractMethod(
    fullFile.readAsStringSync(),
    'void _showEditClientDialog',
  );

  String addClientStr = editClientStr
      .replaceAll(
        '_showEditClientDialog(BuildContext context, Client c)',
        '_showAddClientDialog(BuildContext context)',
      )
      .replaceAll('text: c.name', 'text: ""')
      .replaceAll('text: c.contact.name', 'text: ""')
      .replaceAll('text: c.contact.email', 'text: ""')
      .replaceAll('text: c.contact.phone', 'text: ""')
      .replaceAll('text: c.contact.address', 'text: ""')
      .replaceAll('text: c.contact.website', 'text: ""')
      .replaceAll('text: c.monthlyPayable.toString()', 'text: ""')
      .replaceAll('text: c.nextDueDate', 'text: ""')
      .replaceAll('text: c.paymentsDue.toString()', 'text: "0"')
      .replaceAll('text: c.weeklyReels.toString()', 'text: "0"')
      .replaceAll('text: c.weeklyPosts.toString()', 'text: "0"')
      .replaceAll('text: c.weeklyCarousels.toString()', 'text: "0"')
      .replaceAll('text: c.weeklyStories.toString()', 'text: "0"')
      .replaceAll('text: c.campaigns.toString()', 'text: "0"')
      .replaceAll('text: c.campaignReach', 'text: ""')
      .replaceAll('text: c.postRequirements', 'text: ""')
      .replaceAll(
        'String packageType = c.packageType;',
        'String packageType = "Growth";',
      )
      .replaceAll(
        'String contractPeriod = c.contractPeriod;',
        'String contractPeriod = "3 Months";',
      )
      .replaceAll(
        'String conversionProbability = c.conversionProbability;',
        'String conversionProbability = "Medium";',
      )
      .replaceAll(
        'String retentionHealth = c.retentionHealth;',
        'String retentionHealth = "Good";',
      )
      .replaceAll(
        'String serviceType = c.serviceType;',
        'String serviceType = "Marketing";',
      )
      .replaceAll(
        'bool hasMarketingCommission = c.hasMarketingCommission;',
        'bool hasMarketingCommission = false;',
      )
      .replaceAll(
        'String? marketingExecutiveId = c.marketingExecutiveId;',
        'String? marketingExecutiveId;',
      )
      .replaceAll(
        'String? assignedVideographerId = c.assignedVideographerId;',
        'String? assignedVideographerId;',
      )
      .replaceAll('text: c.sessionRate.toString()', 'text: "0"')
      .replaceAll('if (c.status == \\\'Lead\\\')', 'if (true)')
      .replaceAll('c.id', 'DateTime.now().millisecondsSinceEpoch.toString()')
      .replaceAll('updateClient', 'addClient')
      .replaceAll('updateClientVideographer', '//updateClientVideographer')
      .replaceAll('EDIT CLIENT', 'ADD CLIENT');

  // Since we know the end of the class is now guaranteed clean, we can just insert it before the last '}'
  cfoContent = cfoFile.readAsStringSync();
  int finalBrace = cfoContent.lastIndexOf('}');
  if (finalBrace != -1) {
    cfoContent =
        cfoContent.substring(0, finalBrace) +
        '\n' +
        editClientStr +
        '\n' +
        addClientStr +
        '\n}\n';
  }

  cfoFile.writeAsStringSync(cfoContent);
  print('Fixed CFO dashboard completely');
}
